lib/flagsmith/sdk/models/flags.rb

```diff
@@ -4,7 +4,79 @@ module Flagsmith
   module Flags
     class NotFound < StandardError; end
 
-    # Flag Collection
+    class BaseFlag
+      include Comparable
+
+      attr_reader :enabled, :value, :default
+      
+      def initialize(enabled:, value:, default:)
+        @enabled = enabled
+        @value = value
+        @default = default
+      end
+  
+      def enabled?
+        enabled
+      end
+      
+      alias is_default default
+    end
+
+    class DefaultFlag < BaseFlag
+      def initialize(enabled:, value:)
+        super(enabled: enabled, value: value, default: true)
+      end
+    end
+
+    class Flag < BaseFlag
```

Comment:
This class was moved from `sdk/models/flag.rb` and some of the functionality was extracted to `BaseFlag` (so we could create the `DefaultFlag` class as well. 

The functionality has not been changed.